### PR TITLE
GFG-0105-Scrambled String Added

### LINKED ISSUE #1251: [Documentation Update]: GFG-0105-Scrambled String
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue Description

This function takes a string as input and returns a scrambled version of the string. The function uses a random permutation of the characters in the input string to generate the scrambled output.

### Suggested Change

This documentation update is necessary and beneficial for several reasons:

Clear and concise description: The documentation provides a clear and concise description of the scramble_string function, making it easy for developers to understand and use the function.
Comprehensive coverage: The documentation covers all aspects of the scramble_string function, including the function signature, overview, example, best practices, performance, testing, and dependencies.
Practical examples: The documentation includes practical examples and best practices to help developers use the scramble_string function effectively.
Improved code quality: By providing clear and comprehensive documentation, the code quality is improved, making it easier for developers to maintain and update the code.
Reduced development time: The documentation reduces development time by providing a clear and concise description of the scramble_string function, making it easy for developers to use the function without having to spend time understanding the code.

### Rationale

This documentation update is necessary and beneficial for several reasons:

Clear and concise description: The documentation provides a clear and concise description of the scramble_string function, making it easy for developers to understand and use the function.
Comprehensive coverage: The documentation covers all aspects of the scramble_string function, including the function signature, overview, example, best practices, performance, testing, and dependencies.
Practical examples: The documentation includes practical examples and best practices to help developers use the scramble_string function effectively.
Improved code quality: By providing clear and comprehensive documentation, the code quality is improved, making it easier for developers to maintain and update the code.
Reduced development time: The documentation reduces development time by providing a clear and concise description of the scramble_string function, making it easy for developers to use the function without having to spend time understanding the code.

### Urgency

High

### Acknowledgements

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository
- [X] I have read and followed the Contribution Guidelines
- [X] I have followed the code style guidelines of this project
- [X] I have checked for any existing open issues that my pull request may address
- [X] I have ensured that my changes do not break any existing functionality
- [X] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently
- [X] I have read the resources for guidance listed below
- [X] I have not used AI-generated content (e.g., ChatGPT, other AI tools)
- [X] I have not used content from existing sites (e.g., text, images)
- [X] I have followed all security rules and only shared trusted resources

### PR DESCRIPTION
**Pull request by Damini2004**
## Related Issue

Fix #1251 
## Description

Added Solution.

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [x] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [ ] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [ ] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [ ] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [ ] [Markdown Guide](https://www.markdownguide.org/)
- [ ] [MDX Documentation](https://mdxjs.com/docs/)
- [ ] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [ ] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [ ] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [ ] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

**Comment by Damini2004** [comment]
@Ajay-Dhangar any changes?

File: dsa-solutions/gfg-solutions/Hard/0101-0200/0105-Scrambled-String.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+You don't need to read input or print anything. You only need to complete the function isScramble() which takes two strings S1 and S2 as input and returns a boolean value.
+
+Expected Time Complexity:  $O(N2)$.
+Expected Auxiliary Space: $O(N2)$.
+
+## Constraints
+
+- S1.length = S2.length
+- S1.length<=31
```
Comment (Ajay-Dhangar) [inline dsa-solutions/gfg-solutions/Hard/0101-0200/0105-Scrambled-String.md:57]: replace `S1.length<=31` to

```md
`S1.length<=31`
```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update your clde based on suggestions

**Comment by Damini2004** [comment]
@Ajay-Dhangar Review it again.

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**